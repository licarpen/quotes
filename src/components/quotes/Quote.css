.Quote {
  text-align: center;
  margin: auto;
  font-family: Arial, Helvetica, sans-serif;
  background: lightgray;
  padding: 30px;

  button {
    width: 100px;
    height: 50px;
    background: whitesmoke;
    border: solid 2px black;
  }

  img {
    height: 500px;
    border: 2px black solid;
    padding: 40px;
    background: whitesmoke;
  }
}