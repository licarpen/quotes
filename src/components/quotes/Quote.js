import React from 'react';
import PropTypes from 'prop-types';

const Quote = ({ quote, character, image }) => (
  <>
    <p>{quote} --{character}</p>
    <img src={image}/>
  </>
);

Quote.propTypes = {
  quote: PropTypes.string.isRequired,
  character: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Quote;
