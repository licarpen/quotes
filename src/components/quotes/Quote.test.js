import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Quote quote='test quote' character='test character' image='test image' onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
