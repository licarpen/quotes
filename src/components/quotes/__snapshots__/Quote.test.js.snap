// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component matches a snapshot 1`] = `
<div
  className="Quote"
>
  <button
    onClick={[Function]}
  >
    Get New Quote
  </button>
  <p>
    "
    test quote
     --
    test character
    "
  </p>
  <img
    src="test image"
  />
</div>
`;
